osu.Game.Tournament/Screens/Gameplay/GameplayScreen.cs

```diff
@@ -152,6 +181,42 @@ protected override void CurrentMatchChanged(ValueChangedEvent<TournamentMatch> m
             scheduledOperation?.Cancel();
         }
 
+        private void beatmapChanged(ValueChangedEvent<BeatmapInfo> beatmap)
+        {
+            if (beatmap.NewValue == null || CurrentMatch.Value == null)
+                return;
+
+            var currentSelectedBeatmapMods = CurrentMatch.Value.Round.Value.Beatmaps.FirstOrDefault(b => b.ID == beatmap.NewValue.OnlineBeatmapID)?.Mods;
+            showVideoForModsAndHideAllElse(currentSelectedBeatmapMods);
+        }
+
+        private void showVideoForModsAndHideAllElse(string currentSelectedBeatmapMods)
+        {
+            bool showFallbackVideo = true;
+
+            foreach (var (mods, video) in modSpecificGameplayVideos)
+            {
+                if (mods.Equals(currentSelectedBeatmapMods, StringComparison.OrdinalIgnoreCase))
+                {
+                    video.FadeIn(FADE_DELAY);
+                    showFallbackVideo = false;
+                }
+                else
+                {
+                    video.FadeOut(FADE_DELAY);
+                }
```

Comment:
I imagine this method of fading could not look 100% properly because you'll likely be seeing a degree of blackness added to it (due to the game background peering out from under both backgrounds in transition). This should probably be fading in the new background, and only after that fade is complete, fading out the old one immediately so that the crossfade looks correct.

Reply:
hmm, good point. This does however present a new problem: if I only hide the currently visible video after the new video has been faded in, there is a possibility the current visible video will be above the new one fading in which means you dont see the fade at all. So I'd somehow have to move the new video to the top of whats visible first.

I dont know the framework well enough to know how to do that. I also dont know if there is a potentially way better way of handling this whole video switching thing. I basically just took the idea from the win screen where theres 2 videos (teamwin-blue and teamwin-red) and the alpha gets toggled depending on which one is needed.

I'll do some research into possibilities in the framework but if you could give me some pointers would be great 😄

Reply:
If all of the videos are in one container, theoretically all that would be required to make one of them frontmost is setting `Depth` to something like `-Time.Current` on it (check out `OsuGame.showOverlayAboveOthers()` for an example of this)

Reply:
would this be ok?